### DISCUSSION TITLE
Difference between .swcrc and tsconfig.json

### DISCUSSION THREAD

**Discussion by alebas1** [swc-project/swc#7913] [Q&A]
Hello, I am currently building an api with fastify and typescript just for fun.

I am new to typescript, but I want my dev exp to be as good as possible and I'm trying to use the best and fastest tools.

First of all, is swc meant for replacing `tsc`?

I am used to set some parameters in my tsconfig like `declaration` (to generate `.d.ts` files) and `noUncheckedIndexedAccess`. But these are not available (yet?) in `.swcrc`.

In a project, am I supposed to use swc WITH tsc ? 

Am I supposed to have `.swcrc` and `tsconfig.json` both present? 
> If yes, how do I know I don't have useless configs
> (e.g with `"sourceMaps": true` in `.swcrc` and  `"sourceMap": false` in `tsconfig.json`, the `tsconfig.json` one will get ignored)

And how do I bring my `declaration` and `noUncheckedIndexedAccess` parameters to swc compilation?

Thanks a lot!

**Comment by kwonoj** [comment]
Those are tsc specific options that swc doesn't support. Swc provides transpilation features but does not offer type checking from typescript compiler. For those you need to keep using tsc.
